pkg/app/ops/insightcollector/application_collector.go

```diff
@@ -88,13 +97,7 @@ func (c *applicationDataCollector) listApplications(ctx context.Context) ([]*mod
 
 	for {
 		apps, next, err := c.lister.List(ctx, datastore.ListOptions{
-			Filters: []datastore.ListFilter{
-				{
-					Field:    "Deleted",
-					Operator: datastore.OperatorEqual,
-					Value:    false,
-				},
-			},
+			Filters: []datastore.ListFilter{},
```

Comment:
Nice catch 👍 I think we should add a comment to the head of this `listApplications` function (L93) to explain its behavior. Even if it looks obvious, it helps to mention that we expect this function to return values that way. Something like

```
// listApplications queries the datastore and gets all projects' applications (including disabled/deleted apps)
// to ensure projects with all applications disabled/deleted are listed as well.
```

Reply:
Thanks, added it! [dd971aa](https://github.com/pipe-cd/pipecd/pull/4953/commits/dd971aaa7b8810bacd9470cfd863cce6b775551d)